Source/Generic/ECLogManager.m

```diff
@@ -290,7 +293,11 @@ - (void)startup
 
 	[self loadSettings];
 	[self registerHandlers];
-	[self notifyDelegateOfStartup];
+	
+	// defer the end of the startup until the main runloop is in action - this gives the client time to add a delegate
+	dispatch_async(dispatch_get_main_queue(), ^{
+		[self finishStartup];
+	});
 }
 
```

Comment:
You are right that it's a hack (sort of), but you can't do initWithDelegate here. 

The log manager is a true singleton, and has to be - it is used by the log channel registration, and log channels register themselves when they are first used, which can be before NSApplicationMain has run (actually it can even be before main() has been called).

Thus you can't pass the delegate to the log manager, because you're not responsible for creating it, and the thing that is responsible will call startup before you've had a chance to insert the delegate. Hence the hack.